railties/lib/rails/generators/rails/app/templates/config/environments/development.rb.tt

```diff
@@ -43,6 +43,8 @@ Rails.application.configure do
   config.action_mailer.raise_delivery_errors = false
 
   config.action_mailer.perform_caching = false
+
+  config.action_mailer.define_url_options = { host: "localhost", port: 3000 }
```

Comment:
Even though we can set this to anything, I felt it was important to keep it consistent with what Rails is served on, in an effort to make [previews](https://guides.rubyonrails.org/action_mailer_basics.html#previewing-emails) link to the application.

Reply:
This sounds like a helpful change—thanks! Quick question: The `bin/rails server` command allows a `--port XXXX` flag (and with `bin/dev` you set it via an env var, e.g. `PORT=XXXX bin/dev`). Any reason not to use that value, with a default of `3000`? I ask because I often use that flag when I have multiple rails apps running.

Reply:
That's a great observation. I think we'd somehow need access to the value set in `Rails::Commands::Server::ServerCommand`:

https://github.com/rails/rails/blob/bf8e0f5f611dc67581d35772b773c14e1bae0fb3/railties/lib/rails/commands/server/server_command.rb#L213-L215

I attempted the following implementation, but if `ENV["PORT"]` is not set, it won't match what's passed to the `--port` flag.

```ruby
config.action_mailer.default_url_options = { host: "localhost", port: ENV.fetch("PORT", 3000).to_i }
```

However, [this](Rails::Server::Options.new.parse!(ARGV)[:Port]) seemed to work.

```ruby
config.action_mailer.default_url_options = { host: "localhost", port: Rails::Server::Options.new.parse!(ARGV)[:Port] }
```